include/hffix.hpp

```diff
@@ -449,10 +467,16 @@ class message_writer {
         // Calculate and write out the BodyLength.
         // BodyLength does not include the SOH character after the BodyLength field.
         // BodyLength does not include the SOH character before the CheckSum field.
-        size_t bodylength = next_ - (body_length_ + 7);
-        for(char* b = body_length_ + 5; b >= body_length_; --b) {
-            *b = '0' + (bodylength % 10);
-            bodylength /= 10;
+        if (body_length_) {
```

Comment:
`push_back_header()` will always set `body_length_`, so `if (body_length_)` is equivalent to "has `push_back_header()` been called?" 

I deliberately left `body_length_` uninitialized in the constructor, and instead documented a precondition for this function that says

> `push_back_header()` method has been called

Skipping the `body_length_` initialization is really just a so-tiny-it's-stupid performance optimization, and I understand that a lot of static analysis tools are offended by uninitialized members, so probably this check is a good idea. If the check fails we cannot continue, so let's say `else throw std::logic_error("message_writer.push_back_trailer called before message_writer.push_back.header")`

Reply:
I'll add that `else` statement.

Reply:
I'm sort of thinking that if the compiler can't tell the initialization of ``body_length_`` is a dead store, it should probably be initialized anyway. And, the check in ``push_back_trailer()`` is most likely very cheap, at least compared to all the other checks I added.

I would imagine a much more useful optimization would be to over-estimate the sizes of some of the pushes and have *fewer* tests (to have lower pressure on the branch predictor's buffer)